### PR TITLE
Header revamp + Search bar revamp

### LINKED ISSUE #3424: Header revamp + Search Bar revamp
**Describe the bug**
There is no hierarchy maintained in the header.
For example, on the right side of the bar, we have almost everything, such as a login button, navigational links, a simulator link, a search button, and a user section, among other things.

As a result, the user has no option to select the items in a good. 

**Expected behavior**
We can organise our offerings, features, and user sections in a more organised and efficient manner. And so improve UX.

**Screenshots**
This is the design I've made to make things in order. (Some of the design and UX is inspired from github header)
image

**Device Information [optional]:**
 - OS: All
 - Browser: All
 
**Are you working on this issue? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by Waishnav**
Fixes #3424 

#### Describe the changes you have made in this PR -
- [x] Header re-organize and maintained hierarchy this time. 
(Brand-logo -- Nav-Links -------- Search-Bar -- Simulator -- User section )
- [x] Search bar UX improved and revamp its design.
- [x] User's Profile picture in header (to make things more personalized for user).

### Screenshots of the changes (If any) -
Mobile/Tablet screen           |  Laptop-Screen
:-------------------------:|:-------------------------:
  |   

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR. 

And sorry for the remaking PR. I accidentally force push the master branch in `header-revamp` branch so it automatically get closed because of no-changes.

### CODE REVIEW

**Comment by vedant-jain03** [comment]
Could you resolve the conflicts

File: app/assets/stylesheets/navbar.scss [hunk 1]
```diff
@@ -188,15 +177,161 @@
...
+
+  span {
+    display: none;
+  }
+}
+
+.navbar-profile-section {
+  align-items: center;
+  display: flex;
+  gap: 10px;
```
[the remark this answers is not in the document]
Comment (Waishnav) [reply app/assets/stylesheets/navbar.scss:255]: @vedant-jain03 Hey I didn't get this review message by code climate. can you please elaborate on what I need to do to resolve it?

**Comment by Waishnav** [comment]
> Could you resolve the conflicts

I'm done with the conflict but didn't get the codeclimate review can you please help me with it.

**Comment by nisha-laur** [comment]
@Waishnav @vedant-jain03 I see that this issue is still open . I would like to work on this issue . Can you please assign it to me . Thanks !

**Pull request closed without merging**